typescript-for-haskellers/article.md

```diff
@@ -0,0 +1,1004 @@
+# TypeScript for Haskellers
+
+Web apps are nowadays a mandatory part of every modern application, no matter how small or complex it is.
+From one-click apps that convert pictures to Photoshop, everyone wants fast and easy access to the app, and Web is one of the easiest ways to do that.
+
+In Serokell, we use [TypeScript](https://www.typescriptlang.org/) for writing web pages.
+(You can read our reasoning for that in our [introductory blog post](https://serokell.io/blog/why-typescript).)
+But our main programming language is Haskell. And in this article, we want to explain how Haskell knowledge can help you write TypeScript code.
+
+In this article, we don't have the goal to cover all similarities or differences.
+We also will not say anything about the disadvantages of TypeScript's type system vs. Haskell's or how it works under the hood.
+Our goal is to show how you can use types in TypeScript.
+
+## Why TypeScript?
+
+Historically, any new language that allow you to write web applications translates its code to JavaScript.
+And there are a lot of such languages and language extensions: TypeScript, Kotlin JS, CoffeeScript, Scala JS, Babel, ClojureScript, etc.
+Generally, you can write web applications in almost every language.
+
+Because we like Haskell, maybe we can write our Web application in a functional language (or even in Haskell)? The answer is yes.
+There is strongly-typed PureScript with Haskell-like syntax, purely functional Elm, Reflex framework for Haskell with GHCJS, F# with Fabel.
+
+Unfortunately, these don't give us proper flexibility in working with JS libraries (whose numbers are enormous) and take more time on fixing related problems.
```

Comment:
What do you mean by "related problems"?

Reply:
Related to libraries management \
Choosing what libs are available, somehow import unavailable libs